File: PSBS.WebFrontEnd/psbs-web-frontend/src/lib/chatStore.js
import { create } from "zustand";
export const useChatStore = create((set) => ({
  chatId: null,
  user: null,
  isSupportChat: false,
  isCurrentUserBlocked: false,
  isReceiverBlocked: false,
  ChangeChat: (chatId, user, isSupport) => {
    set({
      chatId,
      user,
      isSupportChat: isSupport,
      isCurrentUserBlocked: false,
      isReceiverBlocked: false,
    });
  },
}));

File: PSBS.AppFlutterFE/psbs_app_flutter/lib/services/chat_store.dart
import 'package:flutter_zustand/flutter_zustand.dart';
import 'package:psbs_app_flutter/models/user.dart';

// Define your store
class UseChatStore extends Store {
  String chatId = "";
  User? user;
  bool isSupportChat = false;
  UseChatStore(chatId, user, isSupportChat) : super(3);
  void changeChat(chatId, user, isSupport) => set({
        chatId: chatId,
        user: user,
        isSupportChat: isSupport,
      });
}
